node/src/main.rs

```diff
@@ -318,10 +328,23 @@ fn async_main() -> impl Future<Item = (), Error = ()> + Send + 'static {
             )).and_then(|_| Ok(())),
     );
 
+    // Create named subgraph provider for resolving subgraph name->ID mappings
+    let named_subgraph_provider = Arc::new(
+        IpfsSubgraphProviderWithNames::init(
+            logger.clone(),
+            Arc::new(subgraph_provider),
+            store.clone(),
+        ).wait()
```

Comment:
This `wait`... can it not cause freezing (e.g. if something inside `init` were to spawn something)?

Reply:
Based on my current level of tokio understanding, I think it's fine... async_main is already running inside the context of a tokio runtime, and tokio can use other threads

@leodasvacas ?